doc/functions/html_cva.rst

```diff
@@ -0,0 +1,173 @@
+``cva``
+=======
+
+`CVA (Class Variant Authority)`_ is a concept from the JavaScript world and used
+by the well-known `shadcn/ui`_.
+The Cva concept is used to render multiple variations of components, applying
+a set of conditions and recipes to dynamically compose CSS class strings (color, size, etc.),
+to create highly reusable and customizable templates. This function work best by being combined with
+`TwigComponent`_
+
+.. note::
+
+    The ``html_cva`` function is part of the ``HtmlExtension`` which is not
+    installed by default. Install it first:
+
+    .. code-block:: bash
+
+        $ composer require twig/html-extra
+
+    Then, on Symfony projects, install the ``twig/extra-bundle``:
+
+This is powered by a ``html_cva()`` Twig
+function where you define ``base`` classes that should always be present and then different
+``variants`` and the corresponding classes:
+
+.. code-block:: html+twig
+
+    {# templates/alert.html.twig #}
+    {% set alert = html_cva({
+        base: 'alert ',
+        variants: {
+            color: {
+                blue: 'bg-blue',
+                red: 'bg-red',
+                green: 'bg-green',
+            },
+            size: {
+                sm: 'text-sm',
+                md: 'text-md',
+                lg: 'text-lg',
+            }
+        }
+    }) %}
+
+    <div class="{{ alert.apply({color, size}, class)) }}">
+        ...
+    </div>
+
+Then use the ``color`` and ``size`` variants to select the classes needed:
+
+.. code-block:: twig
+
+    {# index.html.twig #}
```

Comment:
Probably best to use the twig _function_ here ([recommanded](https://twig.symfony.com/doc/2.x/tags/include.html#include) instead of the _tag_)